api/server/server.go

```diff
@@ -886,17 +886,28 @@ func (s *Server) getImagesGet(eng *engine.Engine, version version.Version, w htt
 	if err := parseForm(r); err != nil {
 		return err
 	}
-	if version.GreaterThan("1.0") {
+
+	useJSON := version.GreaterThan("1.0")
+	if useJSON {
 		w.Header().Set("Content-Type", "application/x-tar")
 	}
-	var job *engine.Job
+
+	imageExportConfig := &graph.ImageExportConfig{
+		Engine:    eng,
+		Outstream: utils.NewWriteFlusher(w),
+	}
 	if name, ok := vars["name"]; ok {
-		job = eng.Job("image_export", name)
+		imageExportConfig.Names = []string{name}
 	} else {
-		job = eng.Job("image_export", r.Form["names"]...)
+		imageExportConfig.Names = r.Form["names"]
 	}
-	job.Stdout.Add(w)
-	return job.Run()
+
+	if err := s.daemon.Repositories().ImageExport(imageExportConfig); err != nil {
+		sf := streamformatter.NewStreamFormatter(useJSON)
```

Comment:
It's the NEW legacy of `postImagesCreate`.
i thought u guys were not pleasant with `Flushed` or `Used` stuff and somehow i was not able to make binary locally due to network problem so i just corrected it in your way:(
I would write a pr to correct `postImagesCreate` then since it is also weird when you type `docker pull scratch` or sth.